Protocol.md

```diff
@@ -296,6 +296,76 @@ The recipient:
 
   These script pub key forms include only standard forms accepted by the wider set of deployed Bitcoin clients in the network, which increase the chances of successful propagation to miners.
 
+# Message segmentation
+
+The communication protocol specified in [BOLT #8](https://github.com/lightningnetwork/lightning-rfc/blob/master/08-transport.md) is based on the noise protocol that has a fixed limit of 65535 bytes for message lengths.
+This can be insufficient for contract with large number of outcomes (and thus adaptor signatures).
+To circumvent this issue, a simple message segmentation protocol is defined, that splits large messages into a `segment_start` message followed by 
+
+## The `segment_start` message
+
+The segment start message indicates the start of a message that is split in several segments.
+
+1. type: 42900 (`segment_start`)
+1. data:
+  * [`u16`:`nb_segments`]
+  * [`bigsize`:`data_len`]
```

Comment:
`data_len` is guaranteed to be a `u16`, so don't use `bigsize`, even if we had <253 bytes which would fit in 1 u8, it's a corner case and can only happen for the last segment anyway. For all but the last segment we're guaranteed to require 3 bytes for bigsize.

Reply:
True, updated to u16